### PR TITLE
update test-cmd to avoid removed beta APIs

### PR DESCRIPTION
**Pull request by deads2k**
Updates test-cmd to use GA API and removes tests of transient v1beta1 functionality that is being removed in 1.22.  Mostly auth APIs. 

related to https://github.com/kubernetes/kubernetes/pull/99840

/kind cleanup
/priority important-soon
@kubernetes/sig-auth-api-reviews 

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/cli-runtime/pkg/resource/crd_finder.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func CRDFromDynamic(client dynamic.Interface) CRDGetter {
 	return func() ([]schema.GroupKind, error) {
 		list, err := client.Resource(schema.GroupVersionResource{
 			Group:    "apiextensions.k8s.io",
-			Version:  "v1beta1",
+			Version:  "v1",
```
Comment (liggitt) [inline staging/src/k8s.io/cli-runtime/pkg/resource/crd_finder.go:38]: this isn't limited to tests, right? does it need to be backported?

File: test/cmd/legacy-script.sh [hunk 2]
```diff
@@ -908,6 +909,15 @@ runTests() {
     fi
   fi
 
+  ############################
+  # Kubectl deprecated APIs  #
+  ############################
+
+  if kube::test::if_supports_resource "${podsecuritypolicies}" ; then
+    record_command run_role_tests
```
Comment (liggitt) [inline test/cmd/legacy-script.sh:917]: should this be `run_deprecated_api_tests`?

**Comment by deads2k** [comment]
tweaked commits, fixed the offending line.  Diff of diff is good

**Comment by eddiezane** [comment]
Failure is confusing. Might be https://github.com/kubernetes/kubernetes/issues/93782?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/101245/pull-kubernetes-conformance-kind-ga-only-parallel/1384177620295356416

https://storage.googleapis.com/k8s-gubernator/triage/index.html?test=LimitRange%20with%20defaults%20and%20ensure%20pod%20has%20those%20defaults%20applied.

/test pull-kubernetes-conformance-kind-ga-only-parallel

**Pull request merged**